Vagrantfile

```diff
@@ -42,22 +43,47 @@ def findip()
 end
 
 Vagrant.configure("2") do |config|
+
   config.vm.box = "opscode-ubuntu-14.04"
   config.vm.box_url = "http://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_ubuntu-14.04_chef-provisionerless.box"
+  config.vm.synced_folder "./", "/var/www/#{$project}", :mount_options => ['dmode=777,fmode=777']
+
   config.omnibus.chef_version = '11.16.0'
 
-  config.vm.provider "virtualbox" do |box|
+  config.vm.provider "virtualbox" do |box, override|
     box.memory = 1024
     box.cpus = 1
+    
+    if RbConfig::CONFIG['host_os'] =~ /mswin|win|mingw/
+      # Allows long filenames in Windows
+      # See: https://github.com/mitchellh/vagrant/issues/1953
+      box.customize ["sharedfolder", "add", :id, "--name", "project", "--hostpath", (("//?/" + File.dirname(__FILE__)).gsub("/","\\"))]
+      
+      # Allow symlinks (Run `vagrant up` from a CMD Prompt with Admin Privs)
+      box.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate/project", "1"]
+      
+      # Dissable the "default" Synced Folder mount (via Vagrant) in favor of VBox
+      override.vm.synced_folder "./", "/var/www/#{$project}", disabled: true
+      
+      # This is the default way the folder is typically created
+      override.vm.provision :shell, inline: "mkdir /var/www/#{$project} -p"
+      override.vm.provision :shell, inline: "mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` project /var/www/#{$project}", run: "always"
+    end
+    
+    # Ensures this only executes on VirtualBox
+    override.vm.provision "chef_solo" do |chef|
+      json = JSON.parse( IO.read("attributes/default.json") )
+      json.store('project', $project)
+      
+      chef.custom_config_path = "./vendor/vagrantfile/Vagrantfile.chef"
```

Comment:
The path has to be relative to the file being parsed.  Since this is a symlink, the effective path is the root of the repository, we can't even query the Ruby interpreter to get the real path that I know of.  I will check, however.